streams/src/test/java/org/apache/kafka/streams/kstream/internals/KStreamFlatMapTest.java

```diff
@@ -86,4 +88,12 @@ public void testFlatMap() {
             assertEquals(expected[i], supplier.theCapturedProcessor().processed().get(i));
         }
     }
+
+    @Test
+    public void testKeyValueMapperResultNotNull() {
+        final KStreamFlatMap<String, Integer, String, Integer> supplier = new KStreamFlatMap<>((key, value) -> null);
+        final Record<String, Integer> record = new Record<>("K", 0, 0L);
+        final Throwable throwable = assertThrows(NullPointerException.class, () -> supplier.get().process(record));
+        assertEquals(throwable.getMessage(), String.format("KeyValueMapper can't return null from mapping the record: %s", record));
```

Comment:
We prefer to use `assertThat()`:

```suggestion
        assertThat(throwable.getMessage(), is(...);
```

Reply:
1. Just curious, why preferred `assertThat` from `hamcrest`? I saw in our code base, different places use `junit` / `hamcrest`. Any rule for guiding when to use which framework? 
2. Other testing in this file are using `junit`. So if I only use `hamcrest` in this single place, this class will import both `junit` and `hamcrest` and mix the using of those two. I think this is a bit ugly LOL. Do you suggest updating all to using `hamcrest`?

Thanks!

Reply:
In Streams, we use `hamcrest` because I think most of us find `assertThat()` more readable. I think other components do not use `hamcrest`. This is just a minor comment and it will not block the merge. Anyways, if you want to use `assertEquals()` you need to swap the parameters, since with `assertEquals()` the first parameter is the expected value and not the actual value.

Reply:
Got it. Thanks for your explanation. I updated to use the assertThat.